python_modules/dagster/dagster/_grpc/client.py

```diff
@@ -201,10 +201,13 @@ def _streaming_query(
                 e, timeout=timeout, custom_timeout_message=custom_timeout_message
             )
 
-    def ping(self, echo: str) -> str:
+    def ping(self, echo: str) -> dict[str, Any]:
```

Comment:
repurposing ping here is a little surprising to me - i thought that was a super lightweight method to determine if a server is running, this makes the response potentially quite a bit larger. Did you consider a new method?

Reply:
The risk is that adding a new method would require handling backcompat for users running a version that doesn't have it.

Reply:
>  this makes the response potentially quite a bit larger

I think as long as we keep this information small it should be ok

> i thought that was a super lightweight method to determine if a server is running

We do also have the `health` endpoint that has its fixed payload of a single status enum
https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_grpc/server.py#L1064-L1065

In my head it makes sense to repurpose `ping` from "hey you up" -> "yes" to "hey you up" -> "ya here is how i'm doin".